static/src/javascripts/projects/common/utils/url.js

```diff
@@ -44,6 +44,23 @@ define([
                 }
             },
 
+            // equivalent to pushQueryString but uses history.replaceState to
+            // overwrite history
+            replaceQueryString: function (params) {
```

Comment:
I've refactored to make the methods share the common code, which should protect a bit against errors. I'm not sure there is much to test here, as it's all pretty imperative. Perhaps that the defaults are applied correctly? But how much value is added?